keras_cv/layers/preprocessing/base_image_augmentation_layer.py

```diff
@@ -0,0 +1,287 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow.tools.docs import doc_controls
+
+from keras_cv.layers.preprocessing import preprocessing_utils as utils
+
+H_AXIS = -3
+W_AXIS = -2
+
+IMAGES = "images"
+LABELS = "labels"
+TARGETS = "targets"
+BOUNDING_BOXES = "bounding_boxes"
+
+
+class BaseImageAugmentationLayer(tf.keras.__internal__.layers.BaseRandomLayer):
```

Comment:
Hey Divya, in this PR will you actually subclass this BaseImageAugmentationLayer from `keras_cv.layers.base_image_augmentation_layer.BaseImageAugmentationLayer` in every subclass under `"keras_cv/layers/preprocessing/*"`?

So you'll replace references in this repo "tf.keras.__internal__.layers.BaseImageAugmentationLayer" with "keras_cv.layers.preprocessing.base_image_augmentation_layer.BaseImageAugmentationLayer".  

Please note the alias "keras_cv.layers.preprocessing.base_image_augmentation_layer.BaseImageAugmentationLayer" is only for internal use, and in the docstrings we should still recommend "keras_cv.layers.BaseImageAugmentationLayer"

Reply:
I have updated the doc string example and also the imports in all the preprocessing layers.